hadoop-tools/hadoop-azure/src/main/java/org/apache/hadoop/fs/azurebfs/services/AbfsClient.java

```diff
@@ -746,14 +747,18 @@ public AbfsRestOperation getPathStatus(final String path, final boolean includeP
     return op;
   }
 
-  public AbfsRestOperation read(final String path, final long position, final byte[] buffer, final int bufferOffset,
-                                final int bufferLength, final String eTag, String cachedSasToken,
+  public AbfsRestOperation read(String path,
+      byte[] buffer,
+      String cachedSasToken,
```

Comment:
What gets added to the ReadRequestParameters vs in this function signature itself?

Should cachedSasToken be a member of the ReadRequestParameters as well?

Let's make sure these conventions are symmetrical across other RequestParameters as well (ie. AppendRequestParameters).

Reply:
This refactoring is to ease the flow of other parameters that Fastpath feature will enforce. Checkstyle has a limit on number of max arguments and we will overrun it when feature code comes in. 
Yes, its inline to AppendRequestParameters, where cachedsas doesnt flow through this instance as its not a parameter to the function of reading/writing.